backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/impl/IndexSchemaFieldNodeComponentRetrievalStrategy.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.search.impl;
+
+import org.hibernate.search.backend.elasticsearch.document.model.impl.ElasticsearchIndexSchemaFieldNode;
+import org.hibernate.search.util.EventContext;
+import org.hibernate.search.util.SearchException;
+
+public interface IndexSchemaFieldNodeComponentRetrievalStrategy<T> {
+
+	T extractComponent(ElasticsearchIndexSchemaFieldNode<?> schemaNode);
+
+	boolean areCompatible(T component1, T component2);
+
+	SearchException createCompatibilityException(String absoluteFieldPath,
+			ElasticsearchIndexSchemaFieldNode<?> schemaNode1, ElasticsearchIndexSchemaFieldNode<?> schemaNode2,
```

Comment:
Given that you're already passing the absolute field path to this method, wouldn't it make more sense to only pass the "components" (of type `T`) instead of the schema nodes here? So that we display the actual components that are not compatible in the error message? That would remove a bit of noise.

Reply:
Right, done.